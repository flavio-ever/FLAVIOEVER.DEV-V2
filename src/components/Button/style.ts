import styled from 'styled-components'
import tw, { css } from 'twin.macro'

export const Container = styled.button(() => [
  tw`
    text-[1rem]
    font-medium

    cursor-pointer
    border-solid
    border
    border-orange-1

    text-orange-1
    rounded-md
    text-center

    px-10
    py-4

    flex
    justify-center
    items-center

    space-x-3
    `,
  tw`
    hover:bg-orange-1
    hover:text-white-1
      `,
  css`
    transition: background-color 0.2s;
  `
])
