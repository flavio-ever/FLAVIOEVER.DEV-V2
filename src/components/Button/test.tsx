import { render, screen } from '@testing-library/react'

import Button from '.'

describe('<Button/>', () => {
  it('should render Button', () => {
    const { container } = render(<Button>Botao teste</Button>)

    expect(
      screen.getByRole('button', {
        name: /Botao teste/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
