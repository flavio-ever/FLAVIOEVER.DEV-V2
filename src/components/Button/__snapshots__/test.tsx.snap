// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render Button 1`] = `
.c0 {
  font-size: 1rem;
  font-weight: 500;
  cursor: pointer;
  border-style: solid;
  border-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgba(255,119,0,var(--tw-border-opacity));
  --tw-text-opacity: 1;
  color: rgba(255,119,0,var(--tw-text-opacity));
  border-radius: 0.375rem;
  text-align: center;
  padding-left: 2.5rem;
  padding-right: 2.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c0 > :not([hidden]) ~ :not([hidden]) {
  --tw-space-x-reverse: 0;
  margin-right: calc(0.75rem * var(--tw-space-x-reverse));
  margin-left: calc(0.75rem * calc(1 - var(--tw-space-x-reverse)));
}

.c0:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(255,119,0,var(--tw-bg-opacity));
  --tw-text-opacity: 1;
  color: rgba(255,255,255,var(--tw-text-opacity));
}

.c0 svg {
  font-size: 1.4rem;
}

<button
  class="c0"
  type="button"
>
  Botao teste
</button>
`
